File: 20100908_prog2_java_jokenpo/src/dojoprog2/Jokenpo.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package dojoprog2;

/**
 *
 * @author casa
 */
import java.util.HashMap;

public class Jokenpo {
    HashMap dicJokenpo;
    public Jokenpo(){
        dicJokenpo = new HashMap();
        HashMap pedra = new HashMap();
        pedra.put("pedra", "empate");
        pedra.put("tesoura", "pedra");
        pedra.put("papel", "papel");
        
        HashMap papel = new HashMap();
        papel.put("papel", "empate");
        papel.put("tesoura", "tesoura");
        papel.put("pedra", "papel");
        
        HashMap tesoura = new HashMap();
        tesoura.put("tesoura", "empate");
        tesoura.put("pedra", "pedra");
        tesoura.put("papel", "tesoura");
        
        dicJokenpo.put("pedra", pedra);
        dicJokenpo.put("papel", papel);
        dicJokenpo.put("tesoura", tesoura);
    }


    public String jogar(String jogada1, String jogada2){
        return (String)((HashMap) (dicJokenpo.get(jogada1))).get(jogada2);
    }
}

File: 20101019_ufjf_python_jokenpo/jokenpo.py
def avalia_jogo(jogada_1, jogada_2):
    vetor = {
        "pedra":
            {"pedra":"empate","papel":"papel","tesoura":"pedra"},
        "papel":
            {"pedra":"papel","papel":"empate","tesoura": "tesoura"},
        "tesoura":
            {"pedra":"pedra", "papel":"tesoura", "tesoura":"empate"}
    }
    return vetor[jogada_1][jogada_2]
